src/order/filter/basic.lean

```diff
@@ -1699,6 +1704,34 @@ instance ultrafilter.has_bind : has_bind ultrafilter := ⟨@ultrafilter.bind⟩
 instance ultrafilter.functor : functor ultrafilter := { map := @ultrafilter.map }
 instance ultrafilter.monad : monad ultrafilter := { map := @ultrafilter.map }
 
+noncomputable def hyperfilter : filter α := ultrafilter_of cofinite
+
+lemma hyperfilter_le_cofinite (hi : set.infinite (@set.univ α)) : @hyperfilter α ≤ cofinite := 
+  (ultrafilter_of_spec (cofinite_ne_bot hi)).1
+
+lemma is_ultrafilter_hyperfilter (hi : set.infinite (@set.univ α)) : is_ultrafilter (@hyperfilter α) := 
+  (ultrafilter_of_spec (cofinite_ne_bot hi)).2
+
+theorem not_hyper_of_finite (hi : set.infinite (@set.univ α)) {s : set α} :
+set.finite s → s ∉ (@hyperfilter α).sets :=
```

Comment:
Please take a look how whitespace is used for other theorems.
The indentation is (usually) as follows:
```lean
theorem name ... : 
  goal :=
proof 
```
The goal is indented, but the proof starts at the same column as the `theorem` keyword, i.e.g it is not indented. 

Another thing is: `set.finite s → ...` and then start the proof with `λ hf` looks confusing. Instead of an implication add `(hf : set.finite s)` to the arguments. I.e.
```lean
theorem not_hyper_of_finite {s : set α} (hi : set.infinite (@set.univ α)) (hf : set.finite s) :
  s ∉ (@hyperfilter α).sets :=
λ hy, ...
```

Another thing is: I think we should add a `infinite α` type class, which states `set.infinite (@set.univ α)`.

Reply:
Also: the theorem names should follow the statement. I.e. name them
`nmem_hyperfilter_of_finite`